rfcs/0127-issues-warnings.md

```diff
@@ -0,0 +1,288 @@
+
+```nix
+config.problemHandlerDefaultMatchers = [
+  { # Wildcard matcher for everything
+    handler = "warn";
+  }
+  { # Match all security warnings on "hello"
+    package = "hello";
+    kind = "insecure";
+    handler = "error";
+  }
+  { # Match all packages affected by the python2 deprecation
+    name = "python2-eol";
+    handler = "error";
+  }
+  { # Has no effect: the default value is higher
+    kind = "insecure";
+    handler = "ignore";
+  }
+  { # Disallowed: use problemHandlers instead
+    pkgName = "hello";
+    name = "CVE1234";
+    handler = "ignore";
+  }
+]
+```
+
+## Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+### Propagation across transitive dependencies
+
+When a package has a problem that `throw`s, all packages that depend on it will fail to evaluate until that problem is ignored or resolved. Most of the time, this is sufficient.
+
+When the problem requires actions on dependents however, it does not sufficiently inform about all packages that need action. Multiple packages may be annotated with the same problem, in that case it should be given a name and the name should be the same across all instances. Other values like the message or the URL list do not need to be the same and may be adapted sensibly.
+
+### Backwards compatbility, Backporting and stable releases
+
+New problems generally should not be added to stable branches if possible, and also not be backported to them, since it may break evaluation. The same rule applies to other changes to a pacakge's `meta` which may generate a problem and thus lead to evaluation failure too. Scenarios where evaluation failure is a desired goal, for example with unfixable security issues, are obviously exempt from this.
+
+### Removal of packages
+
+The plan is to eventually remove packages with long outstanding problems. The details will be part of future work, but at the very least a package must have a problem whose kind defaults to "error" for at least one full release cycle (so that stable users have sufficient time to be warned and intervene).
+
+If a package needs to be removed for some other reason, the problem kind `removal` should be used instead:
+
+```nix
+meta.problems = {
+  removal = {
+    message = "We don't want this in nixpkgs anymore";
```

Comment:
```suggestion
    message = "This package will be removed from Nixpkgs.";
```

Adhering to your own message format suggestions :)

Reply:
TBF I think those formatting guidelines came from a review comment?